src/rhalphalib/sample.py

```diff
@@ -225,7 +225,12 @@ def setParamEffect(self, param: Parameter, effect_up, effect_down=None, scale=No
             effect_up, binning, _ = _to_numpy(effect_up)
             if not np.array_equal(binning, self.observable.binning):
                 raise ValueError("effect_up has incompatible binning with sample %r" % self)
-            zerobins = (self._nominal <= 0.0) | (effect_up <= 0.0)
+            zerobins_nominal = (self._nominal <= 0.0) | np.isclose(self._nominal, 0.0)
+            zerobins_effect = (effect_up <= 0.0) | np.isclose(effect_up, 0.0)
```

Comment:
On the one hand, the default `atol=1e-8` is near enough as to make no difference for any normal situation. But perhaps when there is a very small signal this would cause trouble. On the other hand such a case would cause trouble later anyway during fitting. So probably ok to leave this `isclose`.